web/src/stream_edit.ts

```diff
@@ -397,45 +403,114 @@ function show_stream_email_address_modal(address: string): void {
         ],
     });
 
+    function get_checked_tags(): string[] {
+        const flags: string[] = [];
+        const $checked_checkboxes = $(".copy-email-modal").find("input:checked");
+        $($checked_checkboxes).each(function () {
+            flags.push($(this).attr("id")!);
+        });
+        return flags;
+    }
+
+    function enable_tag_checkbox_change_handler(): void {
+        $("#copy_email_address_modal .tag-checkbox").on("change", () => {
+            const flags = get_checked_tags();
+            address = get_stream_email_address(flags, address);
+            $(".email-address").text(address);
+        });
+    }
+
+    function generate_email_modal_post_render(): void {
+        function update_option_label(sender: User | CurrentUser | Bot): string {
+            if (sender.user_id === people.EMAIL_GATEWAY_BOT.user_id) {
+                return "Email Gateway Bot";
+            } else if (sender.user_id === current_user.user_id) {
+                return $t({defaultMessage: "You"});
+            }
+            return sender.full_name;
+        }
+
+        function get_options(): {
+            name: string;
+            unique_id: number;
+        }[] {
+            const senders: (User | CurrentUser | Bot)[] = [people.EMAIL_GATEWAY_BOT];
+            senders.push(
+                ...stream_data.get_current_user_and_their_bots_with_post_messages_permission(sub),
+            );
+            return senders.map((sender) => ({
+                name: update_option_label(sender),
+                unique_id: sender.user_id,
+            }));
+        }
+
+        function item_click_callback(event: JQuery.ClickEvent, dropdown: tippy.Instance): void {
+            sender_dropdown_widget.render();
+            $(sender_dropdown_widget.widget_selector).trigger("input");
+            dropdown.hide();
+            $("#copy_email_address_modal .tag-checkbox").off("change");
```

Comment:
(comment for Tim)

I'm disabling the "change" event handler here because if not done, it leads to address being updated behind the scene (visibility: hidden). When enabling all checkboxes it leads to address wrapping on multiple line and the div containing the email address increases in size with no text displayed -- which is kinda weird UX.

Setting `display: None` will also lead to shrinking of the container div of the email-address.

I decided to disable the checkboxes first but Alya pointed out that to be not a good UX, so settled with this approach.

Reply:
OK. We should just put a comment explaining this quirk.